Core/Core/Grades/ViewModel/GradeListViewModel.swift

```diff
@@ -21,42 +21,16 @@ import CombineExt
 import CombineSchedulers
 import Foundation
 
-public enum GradeArrangementOptions: Int, CaseIterable {
-    case groupName = 1
-    case dueDate = 2
-
-    var title: String {
-        switch self {
-        case .groupName:
-            return String(localized: "Group", bundle: .core)
-        case .dueDate:
-            return String(localized: "Due Date", bundle: .core)
-        }
-    }
-}
-
 public final class GradeListViewModel: ObservableObject {
     typealias RefreshCompletion = () -> Void
     typealias IgnoreCache = Bool
 
-    enum ViewState: Equatable {
-        case initialLoading
-        case data(GradeListData)
-        case empty(GradeListData)
-        case error
-    }
-
     // MARK: - Dependencies
 
     private let interactor: GradeListInteractor
 
     // MARK: - Output
-    @Published private(set) var isLoaderVisible = false
-    @Published private(set) var courseName: String?
-    @Published private(set) var courseColor: UIColor?
-    @Published private(set) var totalGradeText: String?
-    @Published private(set) var gradeHeaderIsVisible = false
-    @Published private(set) var state: ViewState = .initialLoading
+    @Published private(set) var gradeListUIModel: GradeListUIModel = .init()
```

Comment:
I removed those dependencies and consolidated them into a single model, GradeListUIModel, which solved the issue of re-rendering the view.

Reply:
Could we please rename it to something like this:
```suggestion
    @Published private(set) var config: GradeListViewModelConfig = .init()
```
(no need for `gradeList` prefix in property name, and not using unclear `UIModel` in type name)